File: src/IsbInterop/Tasks/Wrappers/Notice.vb
Imports IsbInterop.Base.Wrappers

Namespace Tasks.Wrappers

  ''' <summary>
  ''' Обертка над INotice.
  ''' </summary>
  Friend Class Notice
    Inherits CustomJob(Of INoticeInfo)
    Implements INotice

    ''' <summary>
    ''' Информация об объекте.
    ''' </summary>
    Public Overrides ReadOnly Property Info As INoticeInfo
      Get
        Dim rcwINoticeInfo = GetRcwObjectInfo()
        Return New NoticeInfo(rcwINoticeInfo, Scope)
      End Get
    End Property

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwINotice">СOM-объект INotice.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwINotice As Object, scope As IScope)
      MyBase.New(rcwINotice, scope)
    End Sub
  End Class
End Namespace

File: src/IsbInterop.CSharp/Tasks/Wrappers/Notice.cs
using IsbInterop.Base.Wrappers;

namespace IsbInterop.Tasks.Wrappers
{
  /// <summary>
  /// Обертка над INotice.
  /// </summary>
  internal class Notice : CustomJob<INoticeInfo>, INotice
  {
    /// <summary>
    /// Получить информацию об объекте.
    /// </summary>
    /// <returns>Информация об объекте.</returns>
    public override INoticeInfo GetInfo()
    {
      var rcwINoticeInfo = GetRcwObjectInfo();
      return new NoticeInfo(rcwINoticeInfo, Scope);
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwINotice">СOM-объект INotice.</param>
    /// <param name="scope">Область видимости.</param>
    public Notice(object rcwINotice, IScope scope) : base(rcwINotice, scope) { }
  }
}
